src/_pytest/assertion/util.py

```diff
@@ -203,30 +204,28 @@ def assertrepr_compare(
 
     summary = f"{left_repr} {op} {right_repr}"
     highlighter = config.get_terminal_writer()._highlight
-
-    explanation = None
+    explanation: list[str] | None
     try:
-        if op == "==":
-            explanation = _compare_eq_any(left, right, highlighter, verbose)
-        elif op == "not in":
-            if istext(left) and istext(right):
+        match (left, op, right):
+            case (_, "==", _):
+                explanation = _compare_eq_any(left, right, highlighter, verbose)
+            case (str(), "not in", str()):
                 explanation = _notin_text(left, right, verbose)
-        elif op == "!=":
-            if isset(left) and isset(right):
-                explanation = ["Both sets are equal"]
-        elif op == ">=":
-            if isset(left) and isset(right):
-                explanation = _compare_gte_set(left, right, highlighter, verbose)
-        elif op == "<=":
-            if isset(left) and isset(right):
-                explanation = _compare_lte_set(left, right, highlighter, verbose)
-        elif op == ">":
-            if isset(left) and isset(right):
-                explanation = _compare_gt_set(left, right, highlighter, verbose)
-        elif op == "<":
-            if isset(left) and isset(right):
-                explanation = _compare_lt_set(left, right, highlighter, verbose)
-
+            case (
+                set() | frozenset(),
```

Comment:
This duplicates the `isset` logic, while that seems unlikely to change it seems conceptually wrong to duplicate since the idea is to have a single place which defines what a set is. WDYT?

Reply:
Also wondering if merging the entire ``_compare_eq_any`` in the match case of ``assertrepr_compare`` is going to make it too big.